test/m1_facilitator/handlers/DeclaredDepositIntentsHandler/handle.test.ts

```diff
@@ -0,0 +1,198 @@
+
+    assert.isOk(
+      depositIntent,
+      `Record must be present for ${record.messageHash} in deposit intent model`,
+    );
+  });
+
+  it('should change status of message from undeclared to declared for'
+    + ' existing message', async (): Promise<void> => {
+    await messageRepository.save(
+      new Message(
+        record.messageHash,
+        MessageType.Deposit,
+        MessageStatus.Undeclared,
+        MessageStatus.Undeclared,
+        '0x0000000000000000000000000000000000000050',
+        new BigNumber(0),
+        new BigNumber(0),
+        new BigNumber(0),
+        record.intentHash,
+      ),
+    );
+
+    await declaredDepositIntentsHandler.handle([record]);
+
+    const message = await messageRepository.get(record.messageHash);
+
+    assert.strictEqual(
+      message && message.sourceStatus,
+      MessageStatus.Declared,
+      'Source status must be declared',
+    );
+
+    assert.strictEqual(
+      message && message.targetStatus,
+      MessageStatus.Undeclared,
+      'Target status must be undeclared',
+    );
+  });
+
+  it('should handle multiple records', async (): Promise<void> => {
+    const record2 = {
+      contractAddress: '0x0000000000000000000000000000000000000060',
+      messageHash: web3utils.sha3('20'),
+      intentHash: web3utils.sha3('100'),
+      tokenAddress: '0x0000000000000000000000000000000000000062',
+      feeGasLimit: new BigNumber(20),
+      feeGasPrice: new BigNumber(40),
+      blockNumber: new BigNumber('100'),
+      beneficiary: '0x0000000000000000000000000000000000000070',
+      amount: new BigNumber(20),
+    };
+
+    const gateway1 = new Gateway(
+      record2.contractAddress,
+      '0x0000000000000000000000000000000000000010',
+      GatewayType.CONSENSUS,
+      '0x0000000000000000000000000000000000000001',
+      '0x0000000000000000000000000000000000000030',
+      new BigNumber(200),
+    );
+    await gatewayRepository.save(
+      gateway1,
+    );
+
+    await declaredDepositIntentsHandler.handle([record, record2]);
+
+    const message = await messageRepository.get(record.messageHash);
+    const message2 = await messageRepository.get(record2.messageHash);
+
+    assert.strictEqual(
+      message && message.sourceStatus,
+      MessageStatus.Declared,
+      `Source status for message ${record.messageHash} must be declared`,
+    );
+
+    assert.strictEqual(
+      message && message.targetStatus,
+      MessageStatus.Undeclared,
+      `Target status for message ${record.messageHash} must be undeclared`,
+    );
+
+    assert.strictEqual(
+      message2 && message2.sourceStatus,
+      MessageStatus.Declared,
+      `Source status for message ${record2.messageHash} must be declared`,
+    );
+
+    assert.strictEqual(
+      message2 && message2.targetStatus,
+      MessageStatus.Undeclared,
+      `Target status for message ${record2.messageHash} must be undeclared`,
+    );
+
```

Comment:
We should also assert for depositIntent1,2. Since the records are new, they will be created.
We can do deepAssertEqual after creating their objects.

Reply:
We would not be able to used `assertDeepEqual` because of `createdAt` and `updatedAt` fields.
I have added `assertDepositIntentRepository` and `assertMessageRepository` methods for assertion.